cpp/oneapi/dal/algo/decision_forest/infer_types.cpp

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/decision_forest/infer_types.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::decision_forest {
+
+class detail::infer_input_impl : public base {
+public:
+    infer_input_impl(const model& trained_model, const table& data)
+            : trained_model(trained_model),
+              data(data) {}
+    model trained_model;
+    table data;
+};
+
+class detail::infer_result_impl : public base {
+public:
+    table prediction;
```

Comment:
```suggestion
    table labels;
```
For consistency with other classifications (see example SVM, [specs ](https://spec.oneapi.com/versions/latest/elements/oneDAL/source/algorithms/nearest_neighbors/knn_classification.html#training-method-brute-force))

Reply:
fixed